packages/react-tree-grid/stories/Virtualization.stories.tsx

```diff
@@ -0,0 +1,243 @@
+  TreeGridRowProvider,
+  TreeGridRowOnOpenChangeData,
+} from '@fluentui-contrib/react-tree-grid';
+import {
+  Button,
+  Menu,
+  MenuItem,
+  MenuList,
+  MenuPopover,
+  MenuTrigger,
+  useEventCallback,
+  HeadlessFlatTreeItemProps,
+  ForwardRefComponent,
+} from '@fluentui/react-components';
+import { isHTMLElement } from '@fluentui/react-utilities';
+
+import {
+  FixedSizeList,
+  FixedSizeListProps,
+  ListChildComponentProps,
+} from 'react-window';
+import { TreeGridProps } from '../src/components/TreeGrid/TreeGrid.types';
+import { ArrowLeft } from '@fluentui/keyboard-keys';
+
+type Item = {
+  children: string;
+  value: string;
+  parentValue?: string;
+};
+const defaultItems: Item[] = [
+  {
+    value: 'flatTreeItem_lvl-1_item-1',
+    children: `Level 1, item 1`,
+  },
+  ...Array.from({ length: 300 }, (_, i) => ({
+    value: `flatTreeItem_lvl-1_item-1--child:${i}`,
+    parentValue: 'flatTreeItem_lvl-1_item-1',
+    children: `Item ${i + 1}`,
+  })),
+  {
+    value: 'flatTreeItem_lvl-1_item-2',
+    children: `Level 1, item 2`,
+  },
+  ...Array.from({ length: 300 }, (_, index) => ({
+    value: `flatTreeItem_lvl-1_item-2--child:${index}`,
+    parentValue: 'flatTreeItem_lvl-1_item-2',
+    children: `Item ${index + 1}`,
+  })),
+];
+
+type FixedSizeTreeGridProps = Omit<TreeGridProps, 'children'> & {
+  listProps: FixedSizeListProps & { ref?: React.Ref<FixedSizeList> };
+};
+
+const FixedSizeTreeGrid: ForwardRefComponent<FixedSizeTreeGridProps> =
+  React.forwardRef((props, ref) => {
+    const { listProps, ...rest } = props;
+    const handleRef = React.useCallback((instance: HTMLElement | null) => {
+      if (instance) {
+        // This element stays between the treegrid and row
+        // Due to accessibility issues this element should have role="none"
+        instance.setAttribute('role', 'none');
+      }
+    }, []);
+    return (
+      <TreeGrid {...rest} ref={ref}>
+        <FixedSizeList
+          {...listProps}
+          innerRef={handleRef}
+          outerRef={handleRef}
+        />
+      </TreeGrid>
+    );
+  });
+
+type FixedSizeTreeGridRowProps = ListChildComponentProps<
+  HeadlessFlatTreeItemProps[]
+>;
+
+const FixedSizeTreeGridRow = React.memo((props: FixedSizeTreeGridRowProps) => {
+  const item = props.data[props.index];
+  const { openItems, requestOpenChange } = useVirtualizationContext();
+  return item.parentValue === undefined ? (
+    <TreeGridRow
+      data-item-id={item.value}
+      aria-level={1}
+      open={openItems.get(item.value) !== undefined}
+      onOpenChange={(ev, data) =>
+        requestOpenChange({ ...data, index: props.index })
+      }
+      style={props.style}
+    >
+      <TreeGridCell header>{item.children}</TreeGridCell>
+      <TreeGridCell aria-colspan={3}>
+        <Button>Header action</Button>
+      </TreeGridCell>
+    </TreeGridRow>
+  ) : (
+    <TreeGridRowProvider
+      value={{
+        open: !!openItems.get(item.parentValue),
+        level: 1,
```

Comment:
```suggestion
        level: 2,
```

should this be the next level?